verifier/src/error.rs

```diff
@@ -33,6 +33,8 @@ pub enum Error {
     QeIdentityExpired,
     /// QE identity not yet valid
     QeIdentityNotYetValid,
+    /// QE identity version mismatch, expecting 2 got {0}
```

Comment:
In other places we've put the expected version in the error in addition to the one that showed up, mostly to prevent the comparison getting out of sync with the error message. Alternatively the error message could just say `Unexpected QE identity version {0}`

Reply:
Updated this, let me know thoughts here as I used named fields.
I know some code in the main MobileCoin repo uses tuple structs, but I feel they're easy to transpose and not likely well tested when/where used.

Reply:
LGTM